stl/inc/flat_map

```diff
@@ -102,27 +98,34 @@ struct _NODISCARD _Clear_flat_map_scope_guard {
 
 // Implementation
 
-template <class _KeyContainer, class _MappedContainer, bool _IsConst>
-class _Flat_map_iterator_Impl {
-public:
-    using _Key_iterator_t = typename _KeyContainer::const_iterator;
-    using _Mapped_iterator_t =
-        conditional_t<_IsConst, typename _MappedContainer::const_iterator, typename _MappedContainer::iterator>;
-    class type {
+enum class _Paring_iterator_kind : unsigned char {

```

Comment:
Should these names say "pairing"? "Paring" is a kind of cutting/trimming, like with an apple.